sov-modules/sov-state/src/storage.rs

```diff
@@ -105,7 +108,13 @@ pub struct GenericStorage<VR: ValueReader> {
     pub(crate) internal_cache: StorageInternalCache,
 }
 
-impl<VR: ValueReader> Storage for GenericStorage<VR> {
+impl<VR: ValueReader + Default> Storage for GenericStorage<VR> {
```

Comment:
Let's remove the `GenericStorage` type and implement `ZkStorage` directly, the same way as `JmtStorage`.